### PR TITLE
Project/01

### PR DESCRIPTION
**Pull request by Anand-Chavan**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,24 +1,127 @@
...
+import components from './Menu/AdminMenu';
+import { AdminLogin } from './Utils/ApiRes';
+import { toast } from 'react-toastify';
+import { UserDetails } from './Utils/interface';
+
+const getUserDetailsById = async (id: number) => {
+  let responseFromApi!: UserDetails;
+  try {
+    const response = await fetch(`http://localhost:3000/user_details/${id}`, {
+      method: 'GET',
```
Comment (Rushikesh-josh) [inline src/App.tsx]: can we move/handle this in slice please

File: src/App.tsx [hunk 2]
```diff
@@ -1,24 +1,127 @@
...
+      headers: {
+        'Content-Type': 'application/json',
```
Comment (Rushikesh-josh) [inline src/App.tsx]: set the common headers using axios-interceptor

File: src/App.tsx [hunk 3]
```diff
@@ -1,24 +1,127 @@
...
+    console.error('Error during loading:', error);
+    toast.error('Something went wrong');
+  }
+  return responseFromApi;
+};
+
 
 function App() {
+  const [content, setContent] = useState<string>('');
+  const [token, setToken] = useState(localStorage.getItem('token'));
```
Comment (Rushikesh-josh) [inline src/App.tsx]: move this in utils

File: src/App.tsx [hunk 4]
```diff
@@ -1,24 +1,127 @@
...
+  const [loginData, setLoginData] = useState<AdminLogin | null>(null);
+  const [loadInUserData, setLoggedInUsedData] = useState<UserDetails | null>(null);
+
+  const handleLogin = (newToken: string, loginData: any) => {
```
Comment (Rushikesh-josh) [inline src/App.tsx]: move this to helper

File: src/App.tsx [hunk 5]
```diff
@@ -1,24 +1,127 @@
...
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
+    <div style={{ display: 'flex', flexDirection: 'column', minHeight: '100vh' }}>
```
Comment (Rushikesh-josh) [inline src/App.tsx:6]: move this JSX into component

File: src/Common/CommonTable.tsx [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+      columns: columnsWithActions,
+      data,
+      initialState: { pageIndex: 0 } as any,
+    },
+    usePagination
+  ) as any;
+
+
+  return (
+    <div className="common-table">
```
Comment (Rushikesh-josh) [inline src/Common/CommonTable.tsx:99]: move this JSX into separate component

File: src/Common/DeleteConfirmation.tsx [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+type DeleteConfirmationModalProps = {
+    show: boolean;
+    onHide: () => void;
+    onDeleteConfirm: () => void | undefined;
+};
+
+const DeleteConfirmation: FC<DeleteConfirmationModalProps> = ({ show, onHide, onDeleteConfirm }) => {
+
+    return (
+        <div className="modal" style={{ display: 'block' }}>
```
Comment (Rushikesh-josh) [inline src/Common/DeleteConfirmation.tsx]: you already have css file please move inline CSS into the file

File: src/Components/Admin/EnrollPatient.tsx [hunk 8]
```diff
@@ -0,0 +1,235 @@
...
+      onClose();
+    } catch (error) {
+      console.error('Error adding Patient:', error);
+    }
+  }
+
+
+  return (
+
+    <div className="modal" style={{ display: 'block' }}>
```
Comment (Rushikesh-josh) [inline src/Components/Admin/EnrollPatient.tsx:174]: move this in separate component

File: src/Components/Admin/EnrollRoom.tsx [hunk 9]
```diff
@@ -0,0 +1,203 @@
...
+            const requestBody = {
+                room: {
+                    room_type: formData.room_type,
+                    description: formData.description,
+                    charges: formData.charges,
+                    capacity: formData.capacity,
+                },
+            };
+
+            const responseForUser = await fetch('http://localhost:3000/rooms', {
```
Comment (Rushikesh-josh) [inline src/Components/Admin/EnrollRoom.tsx:108]: move this fetching login into the slice

File: src/Components/Admin/EnrollRoom.tsx [hunk 10]
```diff
@@ -0,0 +1,203 @@
...
+            onSuccess(responseForUserDetails);
+            toast.success('User Updated successful!');
+            onClose();
+        } catch (error) {
+            console.error('Error updating Room:', error);
+        }
+    };
+
+    return (
+        <div className="modal" style={{ display: 'block' }}>
```
Comment (Rushikesh-josh) [inline src/Components/Admin/EnrollRoom.tsx]: create separate component for JSX

**Review by Rushikesh-josh — COMMENTED** [review]
Please add Redux.
Addressed given review comments throughout the project.
Please use formik for form validation

File: src/Common/CommonTable.tsx (hunk 6 above)
Comment (Anand-Chavan) [reply src/Common/CommonTable.tsx:99]: This is already in seperate common component

File: src/Components/Admin/EnrollPatient.tsx (hunk 8 above)
Comment (Anand-Chavan) [reply src/Components/Admin/EnrollPatient.tsx:174]: Its model , so there is need to move this ?

File: src/Components/Admin/EnrollPatient.tsx [hunk 11]
```diff
@@ -58,19 +83,19 @@ const EnrollPatient: React.FC<EnrollPatientProps> = ({ onClose, onSuccess, mode,
         password: '',
       }
       setFormData(editObj)
+      console.log(rowData, mode,formData);
```
Comment (Rushikesh-josh) [inline src/Components/Admin/EnrollPatient.tsx]: remove this console log

File: src/Components/Admin/EnrollPatient.tsx [hunk 12]
```diff
@@ -58,19 +83,19 @@ const EnrollPatient: React.FC<EnrollPatientProps> = ({ onClose, onSuccess, mode,
...
     }
   }, [])
 
   const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
-    const { name, value } = e.target;
-    setFormData(prevData => ({
-      ...prevData,
-      [name]: value
-    }));
+    // const { name, value } = e.target;
```
Comment (Rushikesh-josh) [inline src/Components/Admin/EnrollPatient.tsx]: remove commented code

File: src/Components/Admin/EnrollPatient.tsx [hunk 13]
```diff
@@ -58,19 +83,19 @@ const EnrollPatient: React.FC<EnrollPatientProps> = ({ onClose, onSuccess, mode,
...
+    // setFormData(prevData => ({
+    //   ...prevData,
+    //   [name]: value
+    // }));
   };
 
-  const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
-    e.preventDefault();
+  const handleSubmit = async (e: any) => {
```
Comment (Rushikesh-josh) [inline src/Components/Admin/EnrollPatient.tsx]: define type instead of any

File: src/Components/Admin/EnrollPatient.tsx [hunk 14]
```diff
@@ -87,17 +112,11 @@ const EnrollPatient: React.FC<EnrollPatientProps> = ({ onClose, onSuccess, mode,
...
-          'Content-Type': 'application/json',
-          'Authorization': localStorage.getItem('token') as string
-        },
-        body: JSON.stringify(requestBody),
-      });
-      const dataForUsers: ApiResponseForUsers = await responseForUser.json();
-      if(dataForUsers.status.errors && dataForUsers.status.errors.length>0){
-        dataForUsers.status.errors.forEach((ele:string)=>{
+
+      const response = await axiosInstance.post(`/users`, JSON.stringify(requestBody));
```
Comment (Rushikesh-josh) [inline src/Components/Admin/EnrollPatient.tsx]: can we move this into slice?

File: src/Components/Admin/EnrollPatient.tsx [hunk 15]
```diff
@@ -110,24 +129,30 @@ const EnrollPatient: React.FC<EnrollPatientProps> = ({ onClose, onSuccess, mode,
...
-      const responseForUserDetails = await fetch('http://localhost:3000/user_details', {
-        method: 'POST',
-        headers: {
-          'Content-Type': 'application/json',
-          'Authorization': localStorage.getItem('token') as string
-        },
-        body: JSON.stringify(requestBodyForUserDetails),
-      });
-      const dataForUserDetails = await responseForUserDetails.json();
+      const responseForUserDetails = await axiosInstance.put(`/user_details`, JSON.stringify(requestBodyForUserDetails));
```
Comment (Rushikesh-josh) [inline src/Components/Admin/EnrollPatient.tsx]: move this into the slice

**Review by Rushikesh-josh — APPROVED** [review]

**Pull request merged**